source/contracts/reporting/ReportingWindow.sol

```diff
@@ -53,20 +53,23 @@ contract ReportingWindow is DelegationTarget, Extractable, ITyped, Initializable
         return true;
     }
 
-    function createMarket(uint256 _endTime, uint8 _numOutcomes, uint256 _numTicks, uint256 _feePerEthInWei, ICash _denominationToken, address _designatedReporterAddress, bytes32 _topic, string _extraInfo) public onlyInGoodTimes afterInitialized payable returns (IMarket _newMarket) {
+    function createMarket(address _sender, uint256 _endTime, uint8 _numOutcomes, uint256 _numTicks, uint256 _feePerEthInWei, ICash _denominationToken, address _designatedReporterAddress) public onlyInGoodTimes afterInitialized payable returns (IMarket _newMarket) {
         require(block.timestamp < startTime);
-        require(universe.getOrCreateReportingWindowByMarketEndTime(_endTime) == this);
-        MarketFactory _marketFactory = MarketFactory(controller.lookup("MarketFactory"));
-        getReputationToken().trustedReportingWindowTransfer(msg.sender, _marketFactory, universe.getOrCacheDesignatedReportNoShowBond());
-        _newMarket = _marketFactory.createMarket.value(msg.value)(controller, this, _endTime, _numOutcomes, _numTicks, _feePerEthInWei, _denominationToken, msg.sender, _designatedReporterAddress);
+        require(universe == IUniverse(msg.sender));
+        _newMarket = createMarketInternal(_endTime, _numOutcomes, _numTicks, _feePerEthInWei, _denominationToken, _sender, _designatedReporterAddress);
```

Comment:
I'm curious if, now that we know the source of the stack-depth bug, we could get rid of the internal call here and have it inlined?  Looking at this method, I don't believe we'll have any stack depth issues if we move sender to later in the signature.

Not a necessary change, but would be nice to have fewer `internal` functions.

Reply:
I tried to put the parameters in the order the function receives them but that actually makes the stack depth error return. This was the magical configuration of parameters that worked,

Reply:
Same with inlining the function